templates/gnark/go.sum

```diff
@@ -1,54 +0,0 @@
-github.com/bits-and-blooms/bitset v1.8.0 h1:FD+XqgOZDUxxZ8hzoBFuV9+cGWY9CslN6d5MS5JVb4c=
```

Comment:
What's the reason for removing this? I'm not as familiar with golang, but you would typical include lock files with a project in JS, python, or haskell.

Reply:
Removed it for the following reasons, but not a hill I'd choose to die on, so add it back in if you feel strongly about it:
1) I do not want to suggest to users that they must include a go.mod with their upload.  In the handful of cases (admittedly non-exhaustive) I've tried in the past, our API prover's behavior is still stable if you remove it
2) My impression was that lock files communicated versions from the first time a program was compiled successfully.  So when I see a `go.mod` or `cargo.lock`, I envision a standalone piece of code, whereas this scaffold is giving someone a starting point that doesn't even have a main executable function

Reply:
OK, I've done some research and `go.sum` is not directly analogous to lock files, it is just for the validation of checksums of the packages.I would still lean towards including it, I think that's useful even if it serves a different purpose.

Off topic from this PR, but responding to the rest for discussion's sake:

> I do not want to suggest to users that they must include a go.mod with their upload. In the handful of cases (admittedly non-exhaustive) I've tried in the past, our API prover's behavior is still stable if you remove it

We might need to overcome challenges with how to integrate it with our own code, but I'm fairly confident that this will eventually lead to problems if we have real users because they'll be pulling in other third-party dependencies other than gnark and the most common ones. Even the way that we handle our Python dependencies in `requirements.txt` in forge for deployment is considered a bad practice now because it's extremely easy for even transient dependency version changes to break things in production. The transition to lock files has been a long and painful one that was motivated by extremely real issues with deploying and running code. If you haven't run into bugs being introduced by patch updates of dependencies this might not seem so visceral, but it is a common and painful thing.

> My impression was that lock files communicated versions from the first time a program was compiled successfully

I would view them more as an exact specification of what you need for something to run correctly. A program that compiles fine one day can easily break or have new bugs if you compile it again a month later without a lock file. Even on the same day because many dependency resolution algorithms are parallelized and non-deterministic. I would actually expect this to be especially important with public circuits if they're going to be audited or reviewed by someone because you can't fully know what code is being used without knowing its exact dependency versions.